iphone/headers/JavaScriptCore/TiBase.h

```diff
@@ -1,155 +0,0 @@
-/**
```

Comment:
So to follow on:

I was mistaken and thought TiBase.h was removed in this PR, but it is not.

However, The native module template uses TiModule.h, which tracing up the stack imports KrollObject.h which swapped from ToTiJS.h to the actual JavaScriptCore.h.

This means that any aliased Ti* types are now gone. These types were actually available to native modules (though I'm not sure how many actually used them directly). As a result, this is a breaking api change for modules, so the api version for iOS modules should be bumped: https://github.com/appcelerator/titanium_mobile/blob/master/package.json#L6

That would mean that native modules would need to be rebuilt/re-released. I'm not sure if they were going to need to be anyways as a result of this change...

Reply:
Did we test that so far? I would reaaaally like to prevent a breaking module change wherever possible. One possible solution could be to leave the bridge file where it is (legacy), but add a deprecation warning for developers to migrate them. I am planning the same for the Swift module effort, which would basically have a "shelled" file that only imports existing API's. This could eventually work out. Should I try that in this PR, a different one or should we break the modules intentionally?

Reply:
@sgtcoolguy I just tested with an existing app (that uses Ti.Facebook v3.8.0) and it works without recompiling the module. I think that unless we actively use TiCore symbols in modules (like `TiValueRef`), we are fine. And since those are supposed to be private API anyway, people should not use it.